packages/react-portal/src/components/Portal/usePortal.ts

```diff
@@ -14,10 +16,17 @@ const mergeProps = makeMergeProps<PortalState>();
  * @param defaultProps - (optional) default prop values provided by the implementing type
  */
 export const usePortal = (props: PortalProps, defaultProps?: PortalProps): PortalState => {
-  const state = mergeProps({ shouldRender: !useIsSSR() }, defaultProps, props);
+  const virtualParentRootRef = React.useRef<HTMLSpanElement>(null);
+  const state = mergeProps({ shouldRender: !useIsSSR(), virtualParentRootRef }, defaultProps, props);
   const fallbackMountNode = usePortalMountNode({ disabled: !!state.mountNode });
 
   state.mountNode = state.mountNode ?? fallbackMountNode;
 
+  React.useEffect(() => {
+    if (state.virtualParentRootRef.current && state.mountNode) {
+      setVirtualParent(state.mountNode, state.virtualParentRootRef.current);
+    }
+  }, [state.virtualParentRootRef, state.mountNode]);
```

Comment:
I'm not entirely sure if `virtualParentRootRef.current` would ever change, since it is _always_ set internally into that span that users will never have access to.

That inacessible slot _always_ renders even during server render in SSR or in the client when the `Portal` can't be rendered.